integration-tests/tests/src/tests/sync/mixed.ts

```diff
@@ -16,28 +16,131 @@
+  });
+
+  return {
+    null: null,
+    bool: true,
+    int: 1,
+    float: 5.0,
+    string: "stringVal",
+    decimal: Realm.BSON.Decimal128.fromString("1234.5678"),
+    objectId: new Realm.BSON.ObjectId("609afc1290a3c1818f04635e"),
+    uuid: new Realm.BSON.UUID("9476a497-60ef-4439-bc8a-52b8ad0d4875"),
+    date: new Date(1620768552979),
+    data: data,
+    obj: ob,
+  };
+};
+
+const getNestedMixedList = (realm: Realm) => {
+  return [...getMixedList(realm), getMixedList(realm), getMixedDict(realm)];
+};
+
+const getNestedMixedDict = (realm: Realm) => {
+  return {
+    ...getMixedDict(realm),
+    innerList: getMixedList(realm),
+    innerDict: getMixedDict(realm),
+  };
 };
-type Value = Realm.Mixed | ((realm: Realm) => Realm.Mixed);
-type ValueTester = (actual: Realm.Mixed, inserted: Realm.Mixed) => void | boolean;
 
 /**
  * The default tester of values.
  * @param actual The value downloaded from the server.
  * @param inserted The value inserted locally before upload.
  */
-function defaultTester(actual: Realm.Mixed, inserted: Realm.Mixed) {
-  expect(actual).equals(inserted);
+function defaultTester(actual: Realm.Mixed, inserted: Realm.Mixed, realm?: Realm) {
+  if (actual instanceof Realm.List) {
+    const insertedVal = inserted as Realm.Mixed[];
+    actual.forEach((item, index) => defaultTester(item, insertedVal[index], realm));
+  } else if (actual instanceof Realm.Dictionary) {
+    const insertedVal = inserted as { [key: string]: any };
+    Object.keys(actual).forEach((key) => defaultTester(actual[key], insertedVal[key], realm));
+  } else if (actual instanceof Realm.BSON.Decimal128) {
+    const insertedVal = inserted as Realm.BSON.Decimal128;
+    expect(actual.bytes.equals(insertedVal.bytes)).equals(true);
+  } else if (actual instanceof Realm.BSON.ObjectID) {
+    const insertedVal = inserted as Realm.BSON.ObjectID;
+    expect(actual.equals(insertedVal)).equals(true);
+  } else if (actual instanceof Realm.BSON.UUID) {
+    const insertedVal = inserted as Realm.BSON.UUID;
+    expect(actual.equals(insertedVal)).equals(true);
+  } else if (actual instanceof Date) {
+    const insertedVal = inserted as Date;
+    expect(actual.getTime() == insertedVal.getTime()).equals(true);
+  } else if (actual instanceof ArrayBuffer) {
+    const actualBinaryView = new Uint8Array(actual);
+    const insertedBynaryView = new Uint8Array(inserted as ArrayBuffer);
+    expect(actualBinaryView.byteLength).equals(insertedBynaryView.byteLength);
+    actualBinaryView.forEach((item, index) => defaultTester(item, insertedBynaryView[index]));
+  } else if (actual instanceof MixedClass && realm) {
+    const insertedVal = realm.objects<MixedClass>("MixedClass").filtered("_id = $0", actual._id)[0];
+    expect(actual._id.equals(insertedVal._id)).equals(true);
+    defaultTester(actual.value, insertedVal.value);
+  } else {
+    expect(actual).equals(inserted);
+  }
+}
+
+async function setupTest(realm: Realm, useFlexibleSync: boolean) {
```

Comment:
Should we rename this function to something more specific like `subscribeIfFlexibleSync()` instead? Currently, that's the only responsibility of the function, so it could make it clearer.